### PR TITLE
[RELEASE] 20170412

### PR DESCRIPTION
**Pull request by ddollar**
## Pull Requests
  - closes #2076 More HTTP Proxy [@mattmanning]
  - closes #2143 use novalue instead of false because so asg doesn't freak out [@ddollar]
  - closes #2144 avoid DescribeStacks call during event logging [@ddollar]
  - closes #2147 BuildRelease sets last release decrypted env in new release [@nzoschke]

## Milestone Release
- [x] Release branch (20170413001210-20170412)
- [x] Pass CI
- [x] Code review
- [ ] Merge into master
- [ ] Close milestone
- [ ] Release master
- [ ] Record release number: 
- [ ] Pass CI
- [ ] Update demo
- [ ] Deploy demo/httpd
- [ ] Deploy demo/rails
- [ ] Update staging
- [ ] Deploy staging/site-staging
- [ ] Deploy staging/console-staging
- [ ] Verbose [release notes](https://github.com/convox/rack/releases)
- [ ] Documentation review
- [ ] Publish release
- [ ] Release CLI

### CODE REVIEW

File: provider/aws/dist/rack.json [hunk 1]
```diff
@@ -1860,7 +1877,7 @@
                 {
                   "Effect": "Allow",
                   "Action": [
-                    "cloudformation:DescribeStacks",
+                    "cloudformation:DescribeStackResources",
```
Comment (nzoschke) [inline provider/aws/dist/rack.json]: Will this trigger throttling errors on the DescribeStackResources API?

File: provider/aws/lambda/cfevents/main.go [hunk 2]
```diff
@@ -155,3 +133,18 @@ func parseMessage(msg string) (Message, error) {
 
 	return m, nil
 }
+
+func getLogGroup(stack string) (string, error) {
+	res, err := cf.DescribeStackResources(&cloudformation.DescribeStackResourcesInput{
```
Comment (nzoschke) [inline provider/aws/lambda/cfevents/main.go]: moving this off of DescribeStacks does seem like an improvement. But do you expect DescribeStackResources to throttle instead?

There are strategies of caching this stuff on the lambda file system if this isn't good enough.

Comment (ddollar) [inline provider/aws/lambda/cfevents/main.go]: From experience DescribeStacks seems particularly vulnerable to throttling (we added some caching around this in Rack)

Long term the right answer is for CF/ECS events to go into SQS and we can have a worker pulling data off and making bigger chunks of logs API calls

Comment (mwarkentin) [inline provider/aws/lambda/cfevents/main.go]: @ddollar @nzoschke Haven't used this myself, but this kind of sounds like what the [blox cluster-state-service](https://github.com/blox/blox/tree/master/cluster-state-service) they announced at reinvent was meant for. Maybe too much for what you're trying to achieve here.

**Comment by mattmanning** [comment]
Docs review:
- HttpProxy rack param is already documented
- #2143 is an internal improvement, doesn't need user-facing docs
- #2144 is a change in AWS API usage to improve performance, doesn't need docs
- https://convox.com/docs/environment/ already correctly states that env vars are encrypted in S3, unencrypted in TaskDefinitions. No doc updates needed for Dynamo change, since that wasn't documented.

**Pull request merged**